contracts/seed/Seed.sol

```diff
@@ -127,132 +135,157 @@ contract Seed {
+        seedAmountSent = seedAmountSent.add((seedAmountSent.mul(uint256(PPM))).mul(_fee).div(PPM100));
+
+        seedRemainder     = seedAmountSent;
+        seedAmountAtStart = seedAmountSent;
     }
 
     /**
       * @dev                     Buy and lock seed tokens.
-      * @param _seedAmount       The amount of seed tokens to buy.
+      * @param _fundingAmount       The amount of seed tokens to buy.
     */
-    function buy(uint256 _seedAmount) public isActive allowedToBuy {
-        //  fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
-        uint256 fundingAmount = (_seedAmount.mul(price)).div(PCT_BASE);
+    function buy(uint256 _fundingAmount) public isActive allowedToBuy returns(uint256, uint256) {
+        // fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
+        uint256 seedAmount = (_fundingAmount.mul(PCT_BASE)).div(price);
 
         // Funding Token balance of this contract;
-        uint256 fundingBalance = fundingToken.balanceOf(address(this));
+        uint256 fundingBalance = fundingCollected;
 
-        //  feeAmount is an amount of fee we are going to get in seedTokens
-        uint256 feeAmount = (_seedAmount.mul(uint256(PPM))).mul(fee).div(PPM100);
+        // feeAmount is an amount of fee we are going to get in seedTokens
+        uint256 feeAmount = (seedAmount.mul(uint256(PPM))).mul(fee).div(PPM100);
 
         // total fundingAmount should not be greater than the hardCap
         require( fundingBalance.
-                  add(fundingAmount).
-                  add((feeAmount.mul(price)).div(PCT_BASE)) <= hardCap,
+                  add(_fundingAmount) <= hardCap,
             "Seed: amount exceeds contract sale hardCap");
 
-        require( seedRemainder >= _seedAmount.add(feeAmount),
-            "Seed: seed distribution exceeded");
+        require( seedRemainder >= seedAmount.add(feeAmount),
+            "Seed: seed distribution would be exceeded");
+
+        fundingCollected = fundingBalance.add(_fundingAmount);
 
         // the amount of seed tokens still to be distributed
-        seedRemainder = (seedRemainder.sub(_seedAmount)).sub(feeAmount);
+        seedRemainder = (seedRemainder.sub(seedAmount)).sub(feeAmount);
 
         // Here we are sending amount of tokens to pay for lock and fee
-        // FundingTokensSent = fundingAmount + fundingFee
-        require(fundingToken.transferFrom(msg.sender, address(this), fundingAmount.
-            add(feeAmount.mul(price).div(PCT_BASE))), "Seed: no tokens");
+        require(fundingToken.transferFrom(msg.sender, address(this), _fundingAmount), "Seed: no tokens");
 
-        if (fundingToken.balanceOf(address(this)) >= softCap) {
+        if (fundingCollected >= softCap) {
             minimumReached = true;
-        } else if (fundingToken.balanceOf(address(this)) >= hardCap) {
+        } else if (fundingCollected >= hardCap) {
```

Comment:
You should remove the `else` here.  both thresholds can be exceeded in a single buy

Reply:
Yes, this changes have been made, will be in the new PR.